Python/pywarpx/diagnostics/data/BaseReader.py

```diff
@@ -0,0 +1,189 @@
+import collections
+import os
+import re
+
+import numpy as np
+import pandas as pd
+
+
+class DataReader():
+    """
+    Base class that all data readers should inherit from.
+    Works only in Cartesian geometry.
+    """
+
+    def __init__(self, run_directory, file_prefix=None):
+        """
+        Parameters
+        ----------
+        run_directory: string
+            Path to the run directory of WarpX.
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
```

Comment:
Does it make sense to make the current directory (`"."`) the default? I believe that would work to use the directory from which python is executed, which would allow a workflow where the simulation input file and analysis file could be in the sam directory, and the user wouldn't need to specify the run directory.